ssl/statem/extensions.c

```diff
@@ -54,13 +52,14 @@ typedef struct extensions_definition_st {
      */
     int (*init)(SSL *s, unsigned int context);
     /* Parse extension sent from client to server */
-    int (*parse_ctos)(SSL *s, PACKET *pkt, int *al);
+    int (*parse_ctos)(SSL *s, PACKET *pkt, X509 *x, size_t chain, int *al);
```

Comment:
No - it means which certificate this is in the chain as presented by the peer. `numcerts` or `xcount` would be wrong. A Certificate message contains multiple certificates, and each one has its own extensions block. Therefore the parse function is called for each one - if there are 3 certs then parse_ctos will be called three times (if the extension exists) with chain set to 0, 1 and 2. It is bounded by 0 <= chain < "number of certificates". Since the number of certificates is a quantity, the correct type is a size_t.